File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/EmployeeDeduction.cs
using HRManagementSystem.Domain.Enums;

namespace HRManagementSystem.Domain.Entities;

public class EmployeeDeduction
{
    public int EmployeeId { get; set; }
    public int DeductionId { get; set; }
    public decimal? Amount { get; set; }
    public bool? IsPercentage { get; set; }
    public RecurrenceType Recurrence { get; set; } = RecurrenceType.OneTime;
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public Employee Employee { get; set; }
    public Deduction Deduction { get; set; }
}

File: frontend/src/app/models/employee-deduction.ts
export enum RecurrenceType {
    OneTime = 1,
    Period = 2,
    Annual = 3,
    Permanent = 4,
}

export interface EmployeeDeductionDto {
    employeeId: number;
    deductionId: number;
    employeeName?: string;
    deductionName?: string;
    amount?: number;
    isPercentage?: boolean;
    effectiveDate?: string;
    recurrence: RecurrenceType;
    startDate?: string;
    endDate?: string;
}

export interface CreateEmployeeDeductionDto {
    employeeId: number;
    deductionId: number;
    amount?: number;
    isPercentage?: boolean;
    recurrence: RecurrenceType;
    startDate?: string;
    endDate?: string;
}

export interface UpdateEmployeeDeductionDto {
    amount?: number;
    isPercentage?: boolean;
    recurrence: RecurrenceType;
    startDate?: string;
    endDate?: string;
}
